src/main/java/com/minecolonies/coremod/client/gui/WindowClipBoard.java

```diff
@@ -0,0 +1,213 @@
+
+    public WindowClipBoard(@Nullable final int colonyId)
+    {
+        super(Constants.MOD_ID + BUILD_TOOL_RESOURCE_SUFFIX);
+        resourceList = findPaneOfTypeByID(WINDOW_ID_LIST_REQUESTS, ScrollingList.class);
+        this.colonyId = colonyId;
+    }
+
+    /**
+     * Called when the window is opened.
+     * Sets up the buttons for either hut mode or decoration mode.
+     */
+    @Override
+    public void onOpened()
+    {
+        resourceList.setDataProvider(() -> getOpenRequests().size(), (index, rowPane) ->
+        {
+            final ImmutableList<IRequest> openRequests = getOpenRequests();
+            if (index < 0 || index >= openRequests.size())
+            {
+                return;
+            }
+
+            final IRequest request = openRequests.get(index);
+            final ItemIcon exampleStackDisplay = rowPane.findPaneOfTypeByID(LIST_ELEMENT_ID_REQUEST_STACK, ItemIcon.class);
+            final List<ItemStack> displayStacks = request.getDisplayStacks();
+
+            if (!displayStacks.isEmpty())
+            {
+                exampleStackDisplay.setItem(displayStacks.get((lifeCount / LIFE_COUNT_DIVIDER) % displayStacks.size()));
+            }
+            else
+            {
+                final Image logo = findPaneOfTypeByID(DELIVERY_IMAGE, Image.class);
+                logo.setVisible(true);
+                logo.setImage(request.getDisplayIcon());
+            }
+
+            rowPane.findPaneOfTypeByID(REQUESTER, Label.class).setLabelText(request.getRequester().getDisplayName(request.getToken()).getFormattedText());
+
+            rowPane.findPaneOfTypeByID(REQUEST_SHORT_DETAIL, Label.class)
+              .setLabelText(request.getShortDisplayString().getFormattedText().replace("§f", ""));
+        });
+    }
+
+    private ImmutableList<IRequest> getOpenRequests()
+    {
+        final ArrayList<IRequest> requests = Lists.newArrayList();
+        final ColonyView view = ColonyManager.getColonyView(colonyId);
+
+        if (view == null)
+        {
+            return ImmutableList.of();
+        }
+
+        final IPlayerRequestResolver resolver = view.getRequestManager().getPlayerResolver();
+        final IRetryingRequestResolver retryingRequestResolver = view.getRequestManager().getRetryingRequestResolver();
+
+        final Set<IToken> requestTokens = new HashSet<>();
+        requestTokens.addAll(resolver.getAllAssignedRequests());
+        requestTokens.addAll(retryingRequestResolver.getAllAssignedRequests());
+
+        requests.addAll(requestTokens.stream().map(view.getRequestManager()::getRequestForToken).filter(Objects::nonNull).collect(Collectors.toSet()));
+
+        final BlockPos playerPos = Minecraft.getMinecraft().player.getPosition();
+        requests.sort(Comparator.comparing((IRequest request) -> request.getRequester().getDeliveryLocation().getInDimensionLocation().getDistance(playerPos.getX(), playerPos.getY(), playerPos.getZ())).thenComparingInt(Object::hashCode));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 238 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)